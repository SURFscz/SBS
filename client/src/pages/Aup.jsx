import React from "react";
import {withRouter} from "react-router-dom";
import I18n from "i18n-js";
import "./Aup.scss";
import Button from "../components/Button";
import {agreeAup} from "../api";
import CheckBox from "../components/CheckBox";


class Aup extends React.Component {

    constructor(props, context) {
        super(props, context);
        this.state = {agreed: false};
    }

    componentDidMount = () => {
        const {currentUser}  = this.props;
        if (currentUser.user_accepted_aup) {
            this.props.history.push("/home");
        }
    }


    agreeWith = () => agreeAup().then(res => {
        this.props.refreshUser(user => {
            const url = new URL(res.location);
            this.props.history.push(url.pathname + url.search);
        });
    });

    render() {
        const {agreed} = this.state;
        const {currentUser, aupConfig} = this.props;
        const url = I18n.locale === "en" ? aupConfig.url_aup_en : aupConfig.url_aup_nl;
        return (
            <div className="mod-aup">
                <h1>{I18n.t("aup.hi", {name: currentUser.given_name || currentUser.name})}</h1>
                <div className="disclaimer">
                    <p dangerouslySetInnerHTML={{__html: I18n.t("aup.info")}}/>
                </div>
                <h2 dangerouslySetInnerHTML={{__html: I18n.t("aup.title")}}/>
                <p className="" dangerouslySetInnerHTML={{__html: I18n.t("aup.disclaimer", {url: url})}}/>
                <div className="terms">
                    <CheckBox name="aup" value={agreed} info={I18n.t("aup.agreeWithTerms")}
                              onChange={() => this.setState({agreed: !agreed})}/>
                </div>
                <Button className="proceed" onClick={this.agreeWith}
                        txt={I18n.t("aup.onward")} disabled={!agreed}/>

            </div>
        )
    }
}

export default withRouter(Aup);