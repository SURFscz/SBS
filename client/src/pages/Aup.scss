@import "../stylesheets/vars.scss";

.mod-aup {
  max-width: $medium;
  width: 100%;
  box-shadow: 0 2px 4px 0 rgba(0, 0, 0, .1);
  margin: 0 auto 25px auto;
  padding: $medium-padding;

  div.intro {
    padding: $medium-padding 0;

    p {
      margin-bottom: 15px;
    }
  }

  div.htmlAup {
    border-radius: $br;
    border: 1px solid $medium-grey;
    max-height: 380px;
    overflow: scroll;
    background-color: $lightest-grey;
    padding: 10px;

    h1 {
      margin: 5px 0;
      font-size: 16px;
      font-weight: bold;
    }

    ul {
      padding: $medium-padding;
      list-style: circle;
      margin-bottom: $medium-padding;
    }
    p {
      margin-bottom: 5px;
    }
  }

  div.download {
    margin: 25px 0;
    display: flex;

    a.pdf {
      margin-left: auto;
    }
  }

  a.proceed {
    margin-top: 25px;
  }
}
