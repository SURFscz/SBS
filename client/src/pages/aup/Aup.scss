@import "../../stylesheets/vars";

.mod-aup {
    max-width: 620px;
    margin: 25px auto 40px auto;
    width: 100%;

    @media (max-width: 680px) {
        padding: 0 15px;
    }

    h1 {
        margin-bottom: 20px;
    }

    div.terms {
        margin: 40px 0;
    }

}
