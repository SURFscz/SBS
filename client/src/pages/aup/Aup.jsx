import React from "react";
import {withRouter} from "react-router-dom";
import I18n from "../../locale/I18n";
import "./Aup.scss";
import Button from "../../components/button/Button";
import {agreeAup} from "../../api";
import CheckBox from "../../components/checkbox/CheckBox";
import {login} from "../../utils/Login";
import SpinnerField from "../../components/redesign/spinner-field/SpinnerField";
import DOMPurify from "dompurify";
import {redirectToProxyLocation} from "../../utils/ProxyAuthz";
import {dictToQueryParams} from "../../utils/QueryParameters";

class Aup extends React.Component {

    constructor(props, context) {
        super(props, context);
        this.state = {agreed: false, loading: true};
    }

    componentDidMount = () => {
        const {currentUser} = this.props;
        if (currentUser.guest) {
            setTimeout(login, 5);
        } else if (currentUser.user_accepted_aup) {
            this.props.history.push("/home");
        } else {
            this.setState({loading: false})
        }
    }

    agreeWith = config => agreeAup().then(res => {
        this.props.refreshUser(() => {
            const item = window.sessionStorage.getItem("interrupt");
            if (item) {
                const interruptDict = JSON.parse(item);
                const params = dictToQueryParams(interruptDict);
                this.props.history.push(`/interrupt?${params}`);
            } else {
                redirectToProxyLocation(res.location, this.props.history, config);
            }
        });
    });

    render() {
        const {agreed, loading} = this.state;
        const {config, currentUser, aupConfig} = this.props;
        const url = I18n.locale === "en" ? aupConfig.url_aup_en : aupConfig.url_aup_nl;
        if (loading) {
            return <SpinnerField/>;
        }
        return (
            <div className="mod-aup">
                <h1>{I18n.t("aup.hi", {name: currentUser.given_name || currentUser.name || currentUser.email || ""})}</h1>
                <p className=""
                   dangerouslySetInnerHTML={{
                       __html: DOMPurify.sanitize(I18n.t("aup.disclaimer", {url: url}),
                           {ADD_ATTR: ['target']})
                   }}/>
                <div className="terms">
                    <CheckBox name="aup" value={agreed} info={I18n.t("aup.agreeWithTerms")}
                              onChange={() => this.setState({agreed: !agreed})}/>
                </div>
                <Button className="proceed" onClick={() => this.agreeWith(config)}
                        txt={I18n.t("aup.onward")} disabled={!agreed}/>

            </div>
        )
    }
}

export default withRouter(Aup);
