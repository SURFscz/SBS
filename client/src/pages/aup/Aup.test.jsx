import React from "react";
import { render } from "@testing-library/react";
import Aup from "./Aup";
import { MemoryRouter } from "react-router-dom";

jest.mock("../../locale/I18n", () => ({
    t: (key, opts) => key,
    locale: "nl"
}));

const mockConfig = { someConfig: true };
const mockAupConfig = { url_aup_nl: "https://example.com/aup-nl", url_aup_en: "https://example.com/aup-en" };
const mockCurrentUser = { guest: false, user_accepted_aup: false, name: "Test User" };
const mockRefreshUser = jest.fn();
const mockHistory = { push: jest.fn() };

describe("Aup", () => {
    it("matches snapshot with default props", () => {
        const { asFragment } = render(
            <MemoryRouter>
                <Aup
                    config={mockConfig}
                    aupConfig={mockAupConfig}
                    currentUser={mockCurrentUser}
                    refreshUser={mockRefreshUser}
                    history={mockHistory}
                />
            </MemoryRouter>
        );
        expect(asFragment()).toMatchSnapshot();
    });

});
