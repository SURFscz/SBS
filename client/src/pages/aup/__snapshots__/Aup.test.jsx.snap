// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Aup matches snapshot with default props 1`] = `
<DocumentFragment>
  <div
    class="mod-aup"
  >
    <h1>
      aup.hi
    </h1>
    <p
      class=""
    >
      aup.disclaimer
    </p>
    <div
      class="terms"
    >
      <div
        class="sds--checkbox-container sds--tooltip-parent checkbox "
      >
        <input
          id="aup"
          type="checkbox"
        />
        <label
          for="aup"
        >
          <span
            class="sds--checkbox--visual"
          >
            <svg
              height="11.9"
              viewBox="0 0 13.6 11.9"
              width="13.6"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m11.9.4-7 9.6-3.3-3.2c-.4-.4-1-.4-1.3 0-.4.4-.4 1 0 1.3l3.2 3.2q.6.6 1.5.6c.9 0 1.1-.3 1.4-.8l7.1-9.6c.3-.4.2-1-.2-1.3-.5-.3-1.1-.2-1.4.2"
              />
            </svg>
          </span>
          <span
            class="checkbox--text"
          >
            aup.agreeWithTerms
          </span>
        </label>
      </div>
    </div>
    <button
      class="sds--btn sds--btn--primary    "
      disabled=""
      type="button"
    >
      <span
        class="textual"
      >
        aup.onward
      </span>
    </button>
  </div>
</DocumentFragment>
`;
